### PR TITLE
Xero WL/Embed: Export Log - 002

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: src/app/core/models/db/accounting-export.model.ts [hunk 1]
```diff
@@ -214,11 +214,42 @@ export class AccountingExportModel {
     return [`https://www-p02.intacct.com/ia/acct/ur.phtml?.r=${expenseGroup.response_logs?.url_id}`, expenseGroup.export_type];
   }
 
+  static constructXeroExportUrlAndType(expenseGroup: ExpenseGroup): [string, string] {
+    let exportRedirection = '';
+    let exportType = '';
+    let exportId = null;
+    let accountId = null;
+    let xeroShortCode = '';
```
Comment (ashwin1111) [inline src/app/core/models/db/accounting-export.model.ts:222]: check old xero app code, we should get xeroShortCode from workspaces/ call

File: src/app/core/models/db/accounting-export.model.ts [hunk 2]
```diff
@@ -229,7 +260,7 @@ export class AccountingExportModel {
       const referenceNumber = this.getFyleReferenceNumber(referenceType, expenseGroup.expenses[0]);
 
       const [url, exportType] = this.constructExportUrlAndType(appName, expenseGroup);
-
+       console.log(appName, expenseGroup);
```
Comment (ashwin1111) [inline src/app/core/models/db/accounting-export.model.ts:263]: remove loggers

File: src/app/integrations/xero/xero-main/xero-export-log/xero-complete-export-log/xero-complete-export-log.component.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    this.isLoading = true;
+    this.offset = offset;
+    this.currentPage = Math.ceil(offset / this.limit) + 1;
+    this.getAccountingExports(this.limit, offset);
+  }
+
+  private getAccountingExports(limit: number, offset:number) {
+
+      this.isLoading = true;
+
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-main/xero-export-log/xero-complete-export-log/xero-complete-export-log.component.ts:101]: remove all empty lines

File: src/app/integrations/xero/xero-main/xero-export-log/xero-export-log-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    component: XeroExportLogComponent,
+    children: [
+      {
+        path: 'complete',
+        component: XeroCompleteExportLogComponent
+      },
+      {
+        path: 'skipped',
+        component: XeroSkippedExportLogComponent
+      }
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-main/xero-export-log/xero-export-log-routing.module.ts:20]: not required, no skip export feature

File: src/app/core/models/db/accounting-export.model.ts (hunk 1 above)
Comment (anishfyle) [reply src/app/core/models/db/accounting-export.model.ts:222]: unable to find the call, for xeroShortCode

File: src/app/core/models/db/accounting-export.model.ts [hunk 5]
```diff
@@ -272,6 +273,10 @@ export class AccountingExportModel {
         expenses: expenseGroup.expenses
       };
   }
+
+  static assignXeroShortCode(xeroShortCode: string){
```
Comment (ashwin1111) [inline src/app/core/models/db/accounting-export.model.ts:278]: needs fixes according to latest discussion, read from local storage

File: src/app/core/services/common/workspace.service.ts [hunk 6]
```diff
@@ -71,4 +73,9 @@ export class WorkspaceService {
   getWorkspaceGeneralSettings(): Observable<any> {
     return this.apiService.get(`/workspaces/${this.getWorkspaceId()}/settings/general/`, {});
   }
+
+  getWorkspaceById(): Observable<XeroWorkspace> {
```
Comment (ashwin1111) [inline src/app/core/services/common/workspace.service.ts:77]: not required

File: src/app/integrations/xero/xero-main/xero-export-log/xero-complete-export-log/xero-complete-export-log.component.ts [hunk 7]
```diff
@@ -96,23 +104,26 @@ export class XeroCompleteExportLogComponent implements OnInit {
...
     this.exportLogService.getExpenseGroups(TaskLogState.COMPLETE, limit, offset, this.selectedDateFilter, null, this.searchQuery).subscribe((accountingExportResponse: ExpenseGroupResponse) => {
         this.totalCount = accountingExportResponse.count;
 
-      const accountingExports: AccountingExportList[] = accountingExportResponse.results.map((accountingExport: ExpenseGroup) =>
-        AccountingExportModel.parseExpenseGroupAPIResponseToExportLog(accountingExport, this.org_id, AppName.XERO)
-      );
-      this.filteredAccountingExports = accountingExports;
-      this.accountingExports = [...this.filteredAccountingExports];
-      this.isLoading = false;
+        this.workspaceService.getWorkspaceById().subscribe((workspace: XeroWorkspace) => {
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-main/xero-export-log/xero-complete-export-log/xero-complete-export-log.component.ts:115]: same, not required

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**